server/providers/challenges/rdeploy-blob/index.ts

```diff
@@ -17,30 +31,89 @@ class RDeployBlobProvider extends EventEmitter implements Provider {
   private _interval: NodeJS.Timeout
   private challenges: Challenge[]
 
+  private ready = false
+  private downloadMap: Map<string, string>
+
   constructor (options: RDeployBlobProviderOptions) {
     super()
     options = {
       updateInterval: 60 * 1000,
-      useGlobalRDeployDirectory: true,
-      // rDeployDirectory: ''
       ...options
     }
 
     this._updateInterval = options.updateInterval
-    this._rDeployDirectory = options.useGlobalRDeployDirectory
-      ? config.rDeployDirectory
-      : options.rDeployDirectory
+    this._rDeployDirectory = path.join(__dirname, '../../../../', options.rDeployDirectory)
     this._interval = setInterval(() => this._update(), this._updateInterval)
-    this._update()
+
+    this.downloadMap = new Map()
+
+    const fileDir = path.join(this._rDeployDirectory, options.rDeployFiles)
+    fs.readdir(fileDir)
+      .then(files => {
+        return Promise.all(
+          files.map(
+            file => {
+              const filePath = path.join(fileDir, file)
+
+              return fs.stat(filePath)
```

Comment:
We got the list of files from a `readdir`, so this `stat` doesnt seem necessary.

Reply:
We need to make sure it's not a directory or it might error